### ISSUE TITLE
Create export builders

### ISSUE THREAD

**Issue by lboxell** [gslab-econ/gslab_python#27]
Create gslab_scons builders for SVN export and gdrive downloads.

This task should involve updating the template appropriately (or creating a new task to do this) with our new build/externals protocol.

It might be best to wait on this task until https://github.com/gslab-econ/admin/issues/78 is finished.

**Comment by lboxell** [comment]
On hold.

**Comment by lboxell** [comment]
Note we probably want a dropbox implementation of this.

**Comment by lboxell** [comment]
See #52 where it looks we decided to go towards using rclone for our cloud storage management.

**Comment by lboxell** [comment]
See companion branch in template for testing https://github.com/gslab-econ/template/tree/export_testing. Currently have a preliminary implementation for rclone with an emitter that does `rclone ls` to obtain the targets. There are some bugs that still need to be worked out.

**Comment by lboxell** [comment]
Current implementation uses `rclone lsl` to get target and create a file manifest to determine whether any of the files in the cloud have changed since the last run (it will re-download if the cloud files have changed size, time stamp, or name). 

Next step: Unit tests.

Other issues added to outline:
 - Add utility for other builders to get their source lists from lists of files in the cloud. I.e., if we download files from a folder in dropbox, those get specified as the target of the download build step, but we'll also want to specify those files as the source of some analysis build step. We should create a way to easily do this. One option would be to use the same `rclone lsl` procedure given a dropbox directory. An alternative might be to use the file_manifest.txt somehow from the download step. 
 - If a scons target gets edited, only downstream changes get modified, the target itself doesn't get rebuilt. This could be a problem if a file was manually edited and we didn't notice. Unless you do a fresh re-build without cache, this error could continue to be propagated. The 'state_of_repo.log' should help catch this to some extent if people are paying attention to it, but this won't always happen.

**Comment by lboxell** [comment]
Continued work on this should probably wait until the discussion from https://github.com/gslab-econ/gslab_python/issues/52#issuecomment-277778135 is finalized.

**Comment by lboxell** [comment]
Closing task.

Summary:
 - Per #52, we are unsure whether we want data download to be part of the scons framework and so we are putting this task on hold until the next template review. 
 - The code is currently written to download data using a yaml externals file and to create a file manifest that tracks whether raw data has changed in size, timestamp or name since the last download. If it has changed, it re-downloads.
 - The remaining step would be to write unit tests for this the current code as of this commit: https://github.com/gslab-econ/gslab_python/commit/967d911395597082d6f9a22fec9cc05e705b55e5
 - Added a note to the admin outline to reconsider this at a later point.

**Comment by gentzkow** [comment]
@lboxell: Let's make sure we remember to revisit this when we next discuss the template. It sounds like the work you did here is valuable and we'll want to incorporate it one way or another.

**Comment by lboxell** [comment]
@gentzkow,

Sounds good. I envision we'll go through the admin outline the next time we do a team review, so that should spark our memory to reconsider this.
